Tangem/UIComponents/TokenItemComponent/TokenItemComponentModel.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  TokenItemComponentModel.swift
+//  Tangem
+//
+//  Created by Andrew Son on 28/04/23.
+//  Copyright © 2023 Tangem AG. All rights reserved.
+//
+
+import Combine
+import BlockchainSdk
+
+typealias WalletModelId = Int
+
+protocol TokenItemInfoProvider: AnyObject {
+    var walletStatePublisher: AnyPublisher<WalletModel.State, Never> { get }
+    var pendingTransactionPublisher: AnyPublisher<(WalletModelId, Bool), Never> { get }
+
+    func balance(for amountType: Amount.AmountType) -> Decimal
+}
+
+protocol PriceChangeProvider: AnyObject {
+    var priceChangePublisher: AnyPublisher<Void, Never> { get }
+
+    func change(for currencyCode: String, in blockchain: Blockchain) -> Double
+}
+
+final class TokenItemComponentModel: ObservableObject, Identifiable {
```

Comment:
```suggestion
final class TokenItemComponentViewModel: ObservableObject, Identifiable {
```
Это же все таки для view моделька )

Reply:
по названиям немного сложно из-за того что одинаковые названия получаются, не стал добавлять `View`, т.к. сама вьюха называется `Component`, чтобы не городить. Не уверен как лучше делать переход со старого дизайна на новый, не хочется старые вьюхи сейчас переименовывать, но тогда надо будет потом новые вьюхи обновлять

Reply:
Ну вообще да, можно тогда будет новые потом переименовать, или новый нэйминг ввести)